It cannot be compiled because of the following errors: 
zfp-zc.c: In function ‘main’:
zfp-zc.c:479:18: error: too many arguments to function ‘ZC_startCmpr’
   dataProperty = ZC_startCmpr(varName, ZC_DOUBLE, (double*)fi, 0, 0, pp, zz, yy, xx);
                  ^~~~~~~~~~~~
In file included from zfp-zc.c:9:0:
/tmp/z-checker-installer/Z-checker/zc-install/include/zc.h:418:18: note: declared here
 ZC_DataProperty* ZC_startCmpr(const char* varName, int dataType, void *oriData, size_t r5, size_t r4, size_t r3, size_t r2, size_t r1);
                  ^~~~~~~~~~~~
zfp-zc.c:481:18: error: too many arguments to function ‘ZC_startCmpr’
   dataProperty = ZC_startCmpr(varName, ZC_FLOAT, (float*)fi, 0, 0, pp, zz, yy, xx);
                  ^~~~~~~~~~~~
In file included from zfp-zc.c:9:0:
/tmp/z-checker-installer/Z-checker/zc-install/include/zc.h:418:18: note: declared here
 ZC_DataProperty* ZC_startCmpr(const char* varName, int dataType, void *oriData, size_t r5, size_t r4, size_t r3, size_t r2, size_t r1);
                  ^~~~~~~~~~~~
In file included from zfp-zc.c:8:0:
zfp-zc.c:555:21: error: ‘zfp_field {aka struct <anonymous>}’ has no member named ‘np’; did you mean ‘nx’?
     np = MAX(field->np, 1u);
                     ^
../include/zfp/macros.h:5:21: note: in definition of macro ‘MAX’
 #define MAX(x, y) ((x) > (y) ? (x) : (y))
                     ^
zfp-zc.c:555:21: error: ‘zfp_field {aka struct <anonymous>}’ has no member named ‘np’; did you mean ‘nx’?
     np = MAX(field->np, 1u);
                     ^
../include/zfp/macros.h:5:33: note: in definition of macro ‘MAX’
 #define MAX(x, y) ((x) > (y) ? (x) : (y))
                                 ^
Makefile:20: recipe for target '../bin/zfp-zc' failed
make: *** [../bin/zfp-zc] Error 1


One reason is Z-checker doesn't support 4D yet. I need to fix it in Z-checker code (https://github.com/CODARcode/Z-checker), but I am not available recently. 
Maybe you want to try it. :)

--- a/zfp-patches/zfp-zc.c
+++ b/zfp-patches/zfp-zc.c
@@ -476,9 +476,9 @@ int main(int argc, char* argv[])
 	  if(ny==1&&nz==1&&np==1)
 		yy = zz = pp = 0;
 	  if(type==zfp_type_double)
-		dataProperty = ZC_startCmpr(varName, ZC_DOUBLE, (double*)fi, 0, 0, pp, zz, yy, xx);
+		dataProperty = ZC_startCmpr(varName, ZC_DOUBLE, (double*)fi, 0, pp, zz, yy, xx);
 	  else
-		dataProperty = ZC_startCmpr(varName, ZC_FLOAT, (float*)fi, 0, 0, pp, zz, yy, xx);
+		dataProperty = ZC_startCmpr(varName, ZC_FLOAT, (float*)fi, 0, pp, zz, yy, xx);
 	  /* allocate buffer for compressed data */
 	  bufsize = zfp_stream_maximum_size(zfp, field);
 	  if (!bufsize) {

